rpcs3/Emu/RSX/Common/FragmentProgramDecompiler.cpp

```diff
@@ -644,6 +644,21 @@ template<typename T> std::string FragmentProgramDecompiler::GetSRC(T src)
 		break;
 	}
 
+	if (apply_precision_modifier && !src.neg)
+	{
+		if constexpr (!std::is_same<T, SRC0>::value)
+		{
+			if (dst.opcode == RSX_FP_OPCODE_MAD)
+			{
+				// Hardware tests show special behavior on MAD operation
+				// Only src0 obeys precision modifier (sat tested)
+				// Results: 1 * 100 + 0  = 100, 1 * 1 + 100 = 100, 100 * 1 + 0 = 1
+				// NOTE: Neg modifier seems to break this rule; 1 * -100 + 0 = -1 not -99
```

Comment:
1 * -100 + 0 = -1 not -99
Why you would expect -99 and not -100

Reply:
It's NOT math it's computing of rsx. UINT32_MAX + 1 is 0 in computing but not in math for example.